assignment5/MidNumber.java

```diff
@@ -0,0 +1,53 @@
+package assignment5;
+
+import java.util.Comparator;
+import java.util.PriorityQueue;
+
+public class MidNumber {
+    private final PriorityQueue<Integer> higher;
+    private final PriorityQueue<Integer> lower;
+
+    public MidNumber() {
+        higher = new PriorityQueue<>(Integer::compareTo);
+        lower = new PriorityQueue<>(Comparator.<Integer>comparingInt(a -> a).reversed());
+    }
+
+    public static void main(String[] args) {
+        MidNumber midNumber = new MidNumber();
+        midNumber.addNumber(6);
+        midNumber.addNumber(4);
+        midNumber.addNumber(6);
+        midNumber.addNumber(5);
+        midNumber.addNumber(2);
+        midNumber.addNumber(3);
+        System.out.println("mid number is: " + midNumber.getMidNumber());
+    }
+
+    public void addNumber(int num) {
+        higher.add(num);
+        if (higher.size() - lower.size() >= 2) {
+            lower.add(higher.poll());
+        }
+
+        // make sure it is ordered correctly
+        if (lower.size() > 0) {
```

Comment:
No needs for this check

Reply:
in case we added the first number then the lower will be empty so we need this check